src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -87,6 +87,18 @@ public static String getContents() {
         return result;
     }
 
+    public static String getHtmlContents() {
+        String result = clipboard.getHtml();
+        if (result == null) {
+            return "";
+        }
+        return result;
+    }
```

Comment:
You could always overload a method with default arguments.

Reply:
Yes, but my question is _why_ default arguments is not supported in the method signature. I might just be overthinking it, lol.

Reply:
For a why you have to ask James Gosling, Brian Goetz or other people at oracle.
I guess it was simply not a thing in 1995 and is not needed today, as you can overload methods.
It's possible that they also opted against it, since in c++ (and the syntax is very much alike) default values for method parameters are possible too since c++11 I think.

Reply:
Default arguments are an indicator for multiple methods. For instance, never pass `Optional` as method parameter. Read more in [Java by Comparison](https://java.by-comparison.com/) - highly recommended!

In the concrete case, passing "switches" to methods to change behavior is an indicator to use overloading. Which we did in this case. - More clear: NEVER use switches to guide behavior (e.g., if in mode html do that, if in mode XML do that, ...), use overloading.

Reply:
> For a why you have to ask James Gosling, Brian Goetz or other people at oracle.

I guess I'll send him a message, when he accepts my connection :trollface: 
<img width="477" alt="image" src="https://github.com/JabRef/jabref/assets/47507403/6c25c52e-657d-49c9-9a9a-48351acf4aeb">

Reply:
> Read more in [Java by Comparison](https://java.by-comparison.com/) - highly recommended!

Thank you for the book recommendation, I'll add that to my list!

> In the concrete case, passing "switches" to methods to change behavior is an indicator to use overloading. Which we did in this case. - More clear: NEVER use switches to guide behavior (e.g., if in mode html do that, if in mode XML do that, ...), use overloading.

Hmmm ... this seems to be a common pattern in other programming languages as well. Intuitively it makes sense as there would be too much logic in the giant switch method. Would you agree with this statement?

Reply:
Take a look here, this is a good summary i like: https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#20-prefer-class-hierarchies-to-tagged-classes
Also you could take a look of the new data oriented programming paradigm supported by the new Java language features like sealed classes and pattern matching, which aims to support programming in a similar direction.